assets/m4/ac_prog_mysql.m4

```diff
@@ -56,17 +56,17 @@ else
 
 	AC_MSG_CHECKING([if mysql works])
 
-	if echo 'SELECT * FROM user' | $MYSQL $user_switch $passwd_switch $host_switch mysql> /dev/null; then
+	if echo 'SELECT * FROM ENGINES;' | $MYSQL $user_switch $passwd_switch $host_switch information_schema> /dev/null; then
```

Comment:
What is triggering this change?
Did you get some permission errors when selecting from mysql.user table?

Reply:
From https://github.com/tango-controls/TangoSourceDistribution/pull/65/commits/50402c9554a58c3703583ba4c624d4e862ae06eb:

```
The default setup of mariadb on debian unstable in version
10.3.21-MariaDB does not allow access to the database named mysql.

Therefore the check always fails.

Using the information_schema database, which according to [1, 2] is
always present, we can fix that. The table ENGINES was choosen as it is
deemed cheap to query.

[1]: https://mariadb.com/kb/en/information-schema/
[2]: https://dev.mysql.com/doc/refman/5.7/en/information-schema.html
```

Reply:
Just doing

```
thomas@debian:~/devel/TangoSourceDistribution/assets/m4$ mysql -u tango -pXXX
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 176
Server version: 10.3.21-MariaDB-2 Debian buildd-unstable

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> SHOW DATABASES;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| tango              |
+--------------------+
2 rows in set (0.000 sec)

MariaDB [(none)]> 

```

shows that also.

Reply:
> From [50402c9](https://github.com/tango-controls/TangoSourceDistribution/commit/50402c9554a58c3703583ba4c624d4e862ae06eb):

Ooops. Sorry, I didn't look at the commit message. Thanks for the explanations.

> ```
> The default setup of mariadb on debian unstable in version
> 10.3.21-MariaDB does not allow access to the database named mysql.

I hope this won't cause any issue with mysql.proc which is storing our stored procedures?

> Therefore the check always fails.
> 
> Using the information_schema database, which according to [1, 2] is
> always present, we can fix that. The table ENGINES was choosen as it is
> deemed cheap to query.
> 
> [1]: https://mariadb.com/kb/en/information-schema/
> [2]: https://dev.mysql.com/doc/refman/5.7/en/information-schema.html
> ```



> 
> 
> Just doing
> 
> ```
> thomas@debian:~/devel/TangoSourceDistribution/assets/m4$ mysql -u tango -pXXX
> Welcome to the MariaDB monitor.  Commands end with ; or \g.
> Your MariaDB connection id is 176
> Server version: 10.3.21-MariaDB-2 Debian buildd-unstable
> 
> Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
> 
> Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
> 
> MariaDB [(none)]> SHOW DATABASES;
> +--------------------+
> | Database           |
> +--------------------+
> | information_schema |
> | tango              |
> +--------------------+
> 2 rows in set (0.000 sec)
> 
> MariaDB [(none)]> 
> ```
> 
> shows that also.

I got a bit confused at some point because when you do this from the mysql program prompt you get an error : 

```
MariaDB [performance_schema]> SELECT * FROM ENGINES;
ERROR 1146 (42S02): Table 'performance_schema.ENGINES' doesn't exist
```

But when you do it using echo piped to mysql client process it seems to work.
So it looks like the proposed change is working.

Reply:
I guess doing something like:
`if echo 'SHOW ENGINES' | $MYSQL $user_switch $passwd_switch $host_switch > /dev/null; then`

would work too.

Reply:
@bourtemb The database is `information_schema` not `performance_schema`.

> I hope this won't cause any issue with mysql.proc which is storing our stored procedures?

```
MariaDB [information_schema]> SELECT SPECIFIC_NAME FROM ROUTINES;
+------------------+
| SPECIFIC_NAME    |
+------------------+
| class_att_prop   |
| class_pipe_prop  |
| class_prop       |
| dev_att_prop     |
| dev_pipe_prop    |
| dev_prop         |
| ds_start         |
| get_dev_list     |
| import_device    |
| import_event     |
| init_history_ids |
| init_tac_tables  |
| obj_prop         |
| proc_release_nb  |
+------------------+
14 rows in set (0.001 sec)
```

compares IMHO quite well to

https://github.com/tango-controls/TangoDatabase/blob/e8f21db2bdae04595f7a72f7a87ed5a29f63d760/stored_proc.sql.in#L1

Reply:
> @bourtemb The database is `information_schema` not `performance_schema`.

Yes! Of course! Sorry! So forget what I wrote...

> 
> > I hope this won't cause any issue with mysql.proc which is storing our stored procedures?
> 
> ```
> MariaDB [information_schema]> SELECT SPECIFIC_NAME FROM ROUTINES;
> +------------------+
> | SPECIFIC_NAME    |
> +------------------+
> | class_att_prop   |
> | class_pipe_prop  |
> | class_prop       |
> | dev_att_prop     |
> | dev_pipe_prop    |
> | dev_prop         |
> | ds_start         |
> | get_dev_list     |
> | import_device    |
> | import_event     |
> | init_history_ids |
> | init_tac_tables  |
> | obj_prop         |
> | proc_release_nb  |
> +------------------+
> 14 rows in set (0.001 sec)
> ```
> 
> compares IMHO quite well to
> 
> https://github.com/tango-controls/TangoDatabase/blob/e8f21db2bdae04595f7a72f7a87ed5a29f63d760/stored_proc.sql.in#L1

Good news!

Reply:
I can't really judge which SQL statement is good or not. I tried to resemble the original statement.